Testnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -353,14 +364,16 @@ private void AddToken(Address to, UInt256 tokenId)
     /// <param name="data">Additional data with no specified format, sent in call to 'to' if it is a contract.</param>
     private void SafeTransferFromInternal(Address from, Address to, UInt256 tokenId, byte[] data)
     {
-        CanTransfer(tokenId);
-        ValidateToken(tokenId);
 
         Address tokenOwner = GetIdToOwner(tokenId);
+
+        CanTransfer(tokenOwner, tokenId);
+        EnsureAddressIsNotEmpty(tokenOwner);
```

Comment:
Empty address check can be removed probably because `CanTransfer` is called prior which should fail with an empty address.